fb/app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class LikesController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    @liked = likeable.likes.find_by(user_id: current_user.id)
+    if !@liked
+      likeable.likes.create(user_id: current_user.id)
+      redirect_back(fallback_location: root_path)
+    else
+      destroy
+    end
+  end
+
+  def destroy
+    @liked ||= likeable.likes.find_by(user_id: current_user.id)
+    @liked.delete
```

Comment:
What is delete of like was unsuccessful? [OPTIONAL]

Reply:
I will also handle this edge case too